subsys/storage/stream/stream_flash.c

```diff
@@ -144,6 +146,27 @@ int stream_flash_buffered_write(struct stream_flash_ctx *ctx, const u8_t *data,
 	}
 
 	if (flush && ctx->buf_bytes > 0) {
+		fill_length = ctx->buf_bytes %
+				 flash_get_write_block_size(ctx->fdev);
+		if (fill_length) {
```

Comment:
This is side note: We should consider adding some API call for returning the 'erased' byte value for the device. It is at least second time when I see that the value is needed and is either read from device (here) or predefined in code, e.g. in mcuboot.